src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -108,11 +115,27 @@ public ApplicationUpdater(ApplicationUpdaterUtility utility, String[] args) {
         proxyProperties = new HashMap<>(7);
         proxyIgnoredHosts = new HashSet<>();
         // System Environment Properties are checked before application arguments.
+        springbootEnvMap = loadSpringBootEnvProperties();
         checkEnvironmentProperties();
         this.args = parseArguments(args);
         detectInfo = new DetectInfoUtility().createDetectInfo();
     }
     
+    private Map<String, String> loadSpringBootEnvProperties() {
+        String spingBootEnvironmentPropertyJson = utility.getSysEnvProperty(SYS_ENV_PROP_SPRING_BOOT);
+        if (!StringUtils.isEmpty(spingBootEnvironmentPropertyJson)) {
+            try {
+                return new ObjectMapper()
+                        .readValue(spingBootEnvironmentPropertyJson, 
+                                new TypeReference<LinkedHashMap<String, String>>() {});
+            } catch (JsonProcessingException ex) {
+                logger.error("{} Self-Update of Detect failed to load Spring Boot Environment arguments due to the following exception.", 
+                        LOG_PREFIX, ex);
+            }
+        }
+        return Collections.EMPTY_MAP;
+    }
+    
```

Comment:
Method to load the key value pairs from Spring Boot Application system property.